pkg/sql/sqlstats/insights/integration/integration_test: TestInsightsIntegrationForContention failed

pkg/sql/sqlstats/insights/integration/integration_test.TestInsightsIntegrationForContention [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12206658?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12206658?buildTab=artifacts#/) on release-23.1 @ [44682b19135a9f0d764b8fdbc61bb70c27855e42](https://github.com/cockroachdb/cockroach/commits/44682b19135a9f0d764b8fdbc61bb70c27855e42):


```
=== RUN   TestInsightsIntegrationForContention
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegrationForContention604836245
    test_log_scope.go:79: use -show-logs to present logs inline
    insights_test.go:570: condition failed to evaluate within 45s: sql: Scan error on column index 3, name "schema_name": converting NULL to string is unsupported
    panic.go:522: -- test log scope end --
--- FAIL: TestInsightsIntegrationForContention (54.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInsightsIntegrationForContention.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32430

--- a/pkg/sql/sqlstats/insights/integration/insights_test.go
+++ b/pkg/sql/sqlstats/insights/integration/insights_test.go
@@ -572,7 +572,7 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 		query,
 		COALESCE(insight.contention, 0::INTERVAL)::FLOAT,
 		COALESCE(sum(txn_contention.contention_duration), 0::INTERVAL)::FLOAT AS durationMs,
-		txn_contention.schema_name,
+		COALESCE(txn_contention.schema_name, ''::STRING)::STRING AS schema_name,
 		txn_contention.database_name,
 		txn_contention.table_name,
 		txn_contention.index_name,

